api/Map.js

```diff
@@ -0,0 +1,209 @@
+import View from 'ol/View.js';
+import VectorSource from 'ol/source/Vector.js';
+import VectorLayer from 'ol/layer/Vector.js';
+import TileLayer from 'ol/layer/Tile.js';
+
+import {get as getProjection} from 'ol/proj.js';
+
+import * as constants from './constants.js';
+
+// FIXME: temporary
+import SwisstopoSource from '@geoblocks/sources/Swisstopo.js';
+import EPSG_21781 from '@geoblocks/sources/EPSG21781.js';
+
+
+class Map {
+
+  /**
+   * @param {Object} options
+   * @property {string} div
+   * @property {ol.Coordinate} center
+   * @property {number} [zoom=10]
+   * TODO: more options
+   */
+  constructor(options) {
+    /**
+     * @private
+     * @type {View}
+     */
+    this.view_ = new View({
+      projection: getProjection(constants.projection || 'EPSG:21781'),
+      resolutions: constants.resolutions,
+      zoom: options.zoom,
+      center: options.center
+    });
+
+    /**
+     * @private
+     * @type {OLMap}
+     */
+    this.map_ = new OLMap({
+      controls: [],
+      target: options.div,
+      view: this.view_
+    });
+
+    // FIXME: temporary
+    const background = new TileLayer({
+      source: new SwisstopoSource({
+        layer: 'ch.swisstopo.pixelkarte-farbe',
+        format: 'image/jpeg',
+        timestamp: 'current',
+        projection: EPSG_21781,
+        crossOrigin: 'anonymous'
+      })
+    });
+    this.map_.addLayer(background);
+
+    /**
+     * @private
+     * @type {VectorSource}
+     */
+    this.vectorSource_ = new VectorSource();
+
+    /**
+     * @private
+     * @type {VectorLayer}
+     */
+    this.vectorLayer_ = new VectorLayer({
+      source: this.vectorSource_
+    });
+
+    this.map_.addLayer(this.vectorLayer_);
+  }
+
+  /**
+   * @param {ol.Coordinate} center
+   * @param {number} zoom
+   */
+  recenter(center, zoom) {
+    this.view_.setCenter(center);
+    this.view_.setZoom(zoom);
+  }
+
+  /**
+   * @param {Object} options
+   * @property {ol.Coordinate} position
+   * @property {string} [icon]
+   * @property {ol.Size} [size]
+   */
+  addMarker(options = {}) {
+    const marker = new Feature({
+      geometry: new Point(options.position ? options.position : this.view_.getCenter())
+    });
+    if (options.icon) {
+      // FIXME: use size?
+      marker.setStyle(new Style({
+        image: new Icon({
+          src: options.icon
+        })
+      }));
+    }
+    this.vectorSource_.addFeature(marker);
+  }
+
+  /**
+   * @param {string} layer
+   * @param {Array.<string>} layer
+   * @param {boolean} [highlight=false]
+   */
+  recenterOnObjects(layer, ids, highlight = false) {
+
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} name
+   * @param {string} url
+   * @param {Object} [options]
+   * @property {Array.<string>} [attr=['title', 'description']]
+   * @property {function()} [success]
+   * @property {function()} [error]
+   */
+  addCustomLayer(type, name, url, options = {}) {
+    if (type === 'text') {
```

Comment:
no `else` ? So perhaps you can avoid this parameter ? (or add a FIXME if you plan to support more types)

Reply:
The old api supports text and gpx but the new version only supports text.
Because we need to be compatible with the old api, the type must stay; I've removed the if